src/Workspaces/Core/Portable/Remote/RemoteHostClientExtensions.cs

```diff
@@ -1,36 +1,100 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
-using System;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Execution;
+using Microsoft.CodeAnalysis.Internal.Log;
 
 namespace Microsoft.CodeAnalysis.Remote
 {
     internal static class RemoteHostClientExtensions
     {
-        [Obsolete]
-        public static Task<RemoteHostClient.Session> CreateCodeAnalysisServiceSessionAsync(
-            this RemoteHostClient client, Solution solution, CancellationToken cancellationToken)
+        /// <summary>
+        /// Create <see cref="RemoteHostClient.Session"/> for the <paramref name="serviceName"/> if possible.
+        /// otherwise, return null.
+        /// 
+        /// Creating session could fail if remote host is not available. one of example will be user killing
+        /// remote host.
+        /// </summary>
+        public static Task<RemoteHostClient.Session> TryCreateServiceSessionAsync(this RemoteHostClient client, string serviceName, CancellationToken cancellationToken)
         {
-            return CreateCodeAnalysisServiceSessionAsync(
-                client, solution, callbackTarget: null, cancellationToken: cancellationToken);
+            return client.TryCreateSessionAsync(serviceName, callbackTarget: null, cancellationToken: cancellationToken);
         }
 
-        [Obsolete]
-        public static Task<RemoteHostClient.Session> CreateCodeAnalysisServiceSessionAsync(
-            this RemoteHostClient client, Solution solution, object callbackTarget, CancellationToken cancellationToken)
+        /// <summary>
+        /// Create <see cref="SessionWithSolution"/> for the <paramref name="serviceName"/> if possible.
+        /// otherwise, return null.
+        /// 
+        /// Creating session could fail if remote host is not available. one of example will be user killing
+        /// remote host.
+        /// </summary>
+        public static Task<SessionWithSolution> TryCreateServiceSessionAsync(this RemoteHostClient client, string serviceName, Solution solution, CancellationToken cancellationToken)
```

Comment:
Why is it important to include this type as part of the signature?  I would expect clients would just get 'Sessions', and the sessions internally would hold onto whatever they needed.  Why does the caller need to know or expect that this is  a SessionWithSolution?